extensions/artifacts/org.wso2.developerstudio.codenvy.ext.appfactory/src/main/java/org/wso2/developerstudio/codenvy/ext/appfactory/server/client/CloudAdminServiceClient.java

```diff
@@ -0,0 +1,206 @@
+/*
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.wso2.developerstudio.codenvy.ext.appfactory.server.client;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.NodeList;
+import org.wso2.developerstudio.codenvy.ext.appfactory.server.exception.AppFactoryServerException;
+import org.wso2.developerstudio.codenvy.ext.appfactory.server.model.UserCredentials;
+
+import javax.xml.namespace.QName;
+import javax.xml.soap.MessageFactory;
+import javax.xml.soap.SOAPBody;
+import javax.xml.soap.SOAPConstants;
+import javax.xml.soap.SOAPElement;
+import javax.xml.soap.SOAPException;
+import javax.xml.soap.SOAPMessage;
+import javax.xml.ws.Dispatch;
+import javax.xml.ws.Service;
+import javax.xml.ws.handler.MessageContext;
+import javax.xml.ws.soap.SOAPBinding;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Handles all the App Cloud tenant related operations
+ */
+public class CloudAdminServiceClient {
+    private static final Logger log = LoggerFactory.getLogger(CloudAdminServiceClient.class);
+
+    private final static String AUTH_ADMIN_SERVICE = "https://identity.cloud.wso2.com/services/AuthenticationAdmin";
```

Comment:
Why have we hard-coded the admin service URLs? We should at least get the host and port from a config file?